lang/elaborator/src/conversion_checking/unify.rs

```diff
@@ -259,6 +394,107 @@ impl Ctx {
     }
 }
 
+#[derive(Debug, Clone)]
+struct PartialRenaming {
+    map: HashMap<Lvl, Lvl>,
+    /// The context of the metavariable (and its solution)
+    metavar_ctx: LevelCtx,
+    /// The metavariable, tracked here for better error messages
+    meta_var: MetaVar,
+    /// The origin of the unification call, tracked here for better error messages
+    while_elaborating_span: Option<Span>,
+}
+
+impl PartialRenaming {
+    #[allow(clippy::vec_box)]
+    fn from_args(
+        constraint_ctx: &LevelCtx,
+        metavar_ctx: &LevelCtx,
+        meta_var: &MetaVar,
+        args: &[Vec<Box<Exp>>],
+        while_elaborating_span: &Option<Span>,
+    ) -> TcResult<Self> {
+        let mut map = HashMap::default();
+
+        for (fst, telescope) in args.iter().enumerate() {
+            for (snd, arg) in telescope.iter().enumerate() {
+                let to_lvl = Lvl { fst, snd };
+                match &**arg {
+                    Exp::Variable(Variable { idx, name, .. }) => {
```

Comment:
I think this is too restrictive: Other allowed expressions are solved metavariables which resolve to a variable or annotated variables.